splunk-otel-android/src/main/java/com/splunk/rum/CustomZipkinEncoder.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright Splunk Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.splunk.rum;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+
+import zipkin2.Span;
+import zipkin2.codec.BytesEncoder;
+import zipkin2.codec.Encoding;
+import zipkin2.internal.JsonCodec;
+import zipkin2.internal.V2SpanWriter;
+import zipkin2.internal.WriteBuffer;
+
+class CustomZipkinEncoder implements BytesEncoder<Span> {
```

Comment:
Can we annotate this class to explain its purpose, and possibly link to the upstream bug/feature request that we are working around?

Reply:
Added a bunch of javadoc. Unfortunately, the zipkin authors have said they will not change this behavior, so they don't consider it a bug...it's just the way the zipkin library works.